public/pages/main/Main.tsx

```diff
@@ -48,16 +48,34 @@ export function Main(props: MainProps) {
   const totalDetectors = adState.totalDetectors;
   const queryParams = getURLQueryParams(props.location);
   const dataSourceId = queryParams.dataSourceId ? queryParams.dataSourceId : '';
-  const existingParams = "from=0&size=20&search=&indices=&sortField=name&sortDirection=asc";  
-  const constructHrefWithDataSourceId = (basePath, existingParams, dataSourceId) => {
-    const fullUrl = `${window.location.origin}${basePath}?${existingParams}`;
-    const url = new URL(fullUrl);
-    url.searchParams.set('dataSourceId', dataSourceId); 
-    
-    return `#${url.pathname}${url.search}`;
+  const existingParams =
+    'from=0&size=20&search=&indices=&sortField=name&sortDirection=asc';
+
+  const constructHrefWithDataSourceId = (
+    basePath,
+    existingParams = '',
+    dataSourceId = ''
+  ) => {
+    const searchParams = new URLSearchParams(existingParams);
+
+    if (dataSourceId) {
+      searchParams.set('dataSourceId', dataSourceId);
+    }
+
+    return `#${basePath}?${searchParams.toString()}`;
   };
-  const dashboardHref = dataSourceId ? `#${APP_PATH.DASHBOARD}?dataSourceId=${dataSourceId}` : `#${APP_PATH.DASHBOARD}`;
-  const overviewHref = dataSourceId ? `#${APP_PATH.OVERVIEW}?dataSourceId=${dataSourceId}` : `#${APP_PATH.OVERVIEW}`;
+
+  // Usage examples:
+  const dashboardHref = constructHrefWithDataSourceId(
+    APP_PATH.DASHBOARD,
+    '',
+    dataSourceId
+  );
+  const overviewHref = constructHrefWithDataSourceId(
+    APP_PATH.OVERVIEW,
+    '',
+    dataSourceId
+  );
```

Comment:
how about just using these directly below instead of creating these extra vars? You already do for detector list page.

Reply:
updated in the latest revision